modules/objdetect/src/qrcode.cpp

```diff
@@ -570,18 +570,18 @@ bool QRDetect::computeTransformationPoints()
     {
         Mat mask = Mat::zeros(bin_barcode.rows + 2, bin_barcode.cols + 2, CV_8UC1);
         uint8_t next_pixel, future_pixel = 255;
-        int count_test_lines = 0, index = cvRound(localization_points[i].x);
-        for (; index < bin_barcode.cols - 1; index++)
+        int count_test_lines = 0, index_c = cvRound(localization_points[i].x), index_r = cvRound(localization_points[i].y);
+        for (; index_c < bin_barcode.cols - 1 && index_r < bin_barcode.rows; index_c++)
```

Comment:
`index_r` is not changed in the loop. So it should be checked only once. In case if the value is rounded up, outside of image the loop is not executed at all. I propose to clamp index_r to image size to ensure out-of-bound access and handle corner cases correctly.

Reply:
also should check negative value of `localization_points`